src/app/modules/activities-management/components/activity-list/activity-list.component.ts

```diff
@@ -1,51 +1,81 @@
 import { Component, OnInit } from '@angular/core';
 import { select, Store } from '@ngrx/store';
 import { Observable } from 'rxjs';
-import { delay } from 'rxjs/operators';
+import { delay, map } from 'rxjs/operators';
 import { getIsLoading } from 'src/app/modules/activities-management/store/activity-management.selectors';
-import { Activity } from '../../../shared/models';
+import { Activity, ActivityFront } from '../../../shared/models';
 import { allActivities } from '../../store';
-import { DeleteActivity, LoadActivities, SetActivityToEdit } from './../../store/activity-management.actions';
+import { ArchiveActivity, LoadActivities, SetActivityToEdit, UnarchiveActivity } from './../../store/activity-management.actions';
 import { ActivityState } from './../../store/activity-management.reducers';
 
-
 @Component({
   selector: 'app-activity-list',
   templateUrl: './activity-list.component.html',
   styleUrls: ['./activity-list.component.scss'],
 })
 export class ActivityListComponent implements OnInit {
-  activities: Activity[] = [];
+  constructor(private store: Store<ActivityState>) {
+    this.isLoading$ = store.pipe(delay(0), select(getIsLoading));
```

Comment:
I have checked the code and it is written 9 months ago by @juanultimate [here](https://github.com/ioet/time-tracker-ui/blob/aaa4913ccb2cdb58f3d084432b6a0f17229bd6c3/src/app/modules/activities-management/components/activity-list/activity-list.component.ts#L26)
I will read more about it to have details.

Reply:
An observation about delay(0):

> To understand the issue you must know that:
> - Javascript is single threaded;
> - Asynchronous events runs in event loop (a.k.a. Micro Task and Macro Task)
> - When Async event happens, it is added to the Event loop;
> - After async event added to the Event loop, Javascript continues with synchronous code;
> - After no synchronous code left, it runs events code from Event loop.
> This Observable would be synchronous if you wouldn't add delay(0)
> When first Subscription happens (subscribing is synchronous code), Observable emits immediately, because it is also synchronous. But if you add delay(0) (similar to setTimeout), Javascript will wait until all synchronous code (all source.subscribe() in this case), are executed. After that it will run asynchronous delay(0)).

[more info](https://stackoverflow.com/questions/56601176/using-delay0-in-combination-with-refcount)